airflow-core/src/airflow/api_fastapi/execution_api/versions/v2025_05_20.py

```diff
@@ -0,0 +1,51 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+from cadwyn import ResponseInfo, VersionChange, convert_response_to_previous_version_for, schema
+
+from airflow.api_fastapi.execution_api.datamodels.taskinstance import TIRunContext
+
+
+class DowngradeUpstreamMapIndexes(VersionChange):
+    """Downgrade the upstream map indexes type for older clients."""
+
+    description = __doc__
+
+    instructions_to_migrate_to_previous_version = (
+        schema(TIRunContext).field("upstream_map_indexes").had(type=dict[str, int] | None),  # type: ignore
+    )
+
+    @convert_response_to_previous_version_for(TIRunContext)  # type: ignore[arg-type]
+    def downgrade_upstream_map_indexes(response: ResponseInfo = None) -> None:  # type: ignore
+        """
+        Downgrades the `upstream_map_indexes` field when converting to the previous version.
+
+        Ensures that the field is only a dictionary of  [str, int] (old format).
+        """
+        resp = response.body["upstream_map_indexes"]
```

Comment:
```suggestion
        resp = response.body.get("upstream_map_indexes")
```
as the body field is not required I think

Reply:
Can be done, but the body is required: https://github.com/apache/airflow/blob/main/airflow-core/src/airflow/api_fastapi/execution_api/routes/task_instances.py#L96

Reply:
Body is requires, but `upstream_map_indexes` is not required. It has a default of `= None`, which means in the JSON it might not be there.

Reply:
More specifically, in the API spec it is not marked as a required field.

Reply:
Makes sense. Let me do that